cli/boilerplate_cli.go

```diff
@@ -0,0 +1,147 @@
+package cli
+
+import (
+	"github.com/urfave/cli"
+	"fmt"
+	"github.com/gruntwork-io/boilerplate/config"
+	"github.com/gruntwork-io/boilerplate/templates"
+	"github.com/gruntwork-io/boilerplate/util"
+)
+
+// Customize the --help text for the app so we don't show extraneous info
+const CUSTOM_HELP_TEXT = `NAME:
+   {{.Name}} - {{.Usage}}
+
+USAGE:
+   {{if .UsageText}}{{.UsageText}}{{else}}{{.HelpName}} {{if .VisibleFlags}}[global options]{{end}}{{if .Commands}} command [command options]{{end}} {{if .ArgsUsage}}{{.ArgsUsage}}{{else}}[arguments...]{{end}}{{end}}
+   {{if .VisibleFlags}}
+OPTIONS:
+   {{range .VisibleFlags}}{{.}}
+   {{end}}{{end}}{{if len .Authors}}
+EXAMPLES:
+   Generate a project in ~/output from the templates in ~/templates:
+
+       boilerplate --template-folder ~/templates --output-folder ~/output
+
+   Generate a project in ~/output from the templates in ~/templates, using variables passed in via the command line:
+
+       boilerplate --template-folder ~/templates --output-folder ~/output --var "Title=Boilerplate" --var "ShowLogo=false"
+
+   Generate a project in ~/output from the templates in ~/templates, using variables read from a file:
+
+       boilerplate --template-folder ~/templates --output-folder ~/output --var-file vars.yml
+
+   {{if .Version}}{{if not .HideVersion}}
+VERSION:
+   {{.Version}}
+   {{end}}{{end}}
+AUTHOR(S):
+   {{range .Authors}}{{.}}{{end}}
+   {{end}}{{if .Copyright}}
+COPYRIGHT:
+   {{.Copyright}}
+   {{end}}
+`
+
+func CreateBoilerplateCli(version string) *cli.App {
+	cli.AppHelpTemplate = CUSTOM_HELP_TEXT
+	app := cli.NewApp()
+
+	app.Name = "boilerplate"
+	app.Author = "Gruntwork <www.gruntwork.io>"
+	app.Usage = "A tool for generating files and folders (\"boilerplate\") from a set of templates"
+	app.UsageText = "boilerplate [OPTIONS]"
+	app.Version = version
+	app.Action = runApp
+
+	app.Flags = []cli.Flag {
+		cli.StringFlag{
+			Name: config.OPT_TEMPLATE_FOLDER,
+			Usage: "Generate the project from the templates in `FOLDER`.",
```

Comment:
This is supported by the CLI library we're using. See here for docs: https://github.com/urfave/cli#placeholder-values